x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java

```diff
@@ -79,34 +81,83 @@ protected void masterOperation(
             EnterpriseSearchFeatureSetUsage usage = new EnterpriseSearchFeatureSetUsage(
                 LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
                 enabled,
+                Collections.emptyMap(),
                 Collections.emptyMap()
             );
             listener.onResponse(new XPackUsageFeatureResponse(usage));
             return;
         }
 
-        try {
-            ListSearchApplicationAction.Response resp = clientWithOrigin.execute(
-                ListSearchApplicationAction.INSTANCE,
-                new ListSearchApplicationAction.Request(null, new PageParams(0, 0))
-            ).get();
+        Map<String, Object> searchApplicationsUsage = new HashMap<>();
+        Map<String, Object> analyticsCollectionsUsage = new HashMap<>();
+
+        // Step 2: Fetch search applications count and return usage
+        ListSearchApplicationAction.Request searchApplicationsCountRequest = new ListSearchApplicationAction.Request(
+            "*",
+            new PageParams(0, 0)
+        );
+        ActionListener<ListSearchApplicationAction.Response> searchApplicationsCountListener = ActionListener.wrap(response -> {
+            addSearchApplicationsUsage(response, searchApplicationsUsage);
             listener.onResponse(
                 new XPackUsageFeatureResponse(
                     new EnterpriseSearchFeatureSetUsage(
-                        enabled,
                         LicenseUtils.LICENSED_ENT_SEARCH_FEATURE.checkWithoutTracking(licenseState),
-                        Map.of("count", resp.queryPage().count())
+                        enabled,
+                        searchApplicationsUsage,
+                        analyticsCollectionsUsage
                     )
                 )
             );
-        } catch (Exception e) {
-            logger.warn("Failed to get search application count to include in Enterprise Search usage", e);
```

Comment:
should we still do some error handling here?

Reply:
what do you mean? the error is handled on line 111.

if we throw any errors in this method we will break all of xpack/usage